sklearn/feature_selection/tests/test_variance_threshold.py

```diff
@@ -26,3 +28,17 @@ def test_variance_threshold():
     for X in [data, csr_matrix(data)]:
         X = VarianceThreshold(threshold=.4).fit_transform(X)
         assert_equal((len(data), 1), X.shape)
+
+
+def test_zero_variance_floating_point_error():
+    # Test that VarianceThreshold(0.0).fit eliminates features that have
+    # the same value in every sample, even when floating point errors
+    # cause np.var not to be 0 for the feature.
+    # See #13691
+
+    data = [[-0.13725701]] * 10
+    assert_not_equal(np.var(data), 0.)
+    for X in [data, csr_matrix(data), csc_matrix(data), bsr_matrix(data)]:
+        assert_raises(ValueError, VarianceThreshold().fit, X)
```

Comment:
why should this raise a ValueError?

Reply:
The variance of the single feature is zero, so it should raise `ValueError: No feature in X meets the variance threshold 0.00000` (same as the last case in `test_zero_variance` I think).

Reply:
Oh, right, you're testing that all features are removed. I think that deserves a comment, or at least to test that the ValueError message you expect is matched.

Reply:
please use

```py
msg = ...
with pytest.raises(ValueError, match=msg):
    VarianceThreshold().fit(X)